### PR TITLE
Altered codebase to use orjson

### PR DESCRIPTION
**Pull request by owencjones**
Replaced usage of Python `json` standard lib, with `orjson`, a C-compiled, speed-optimised lib.

Also tested ujson, but orjson had native support for Numpy, and so worked better

# Changelog
* Added orjson to dependencies
* Replaced all uses of `json` with `orjson`

### CODE REVIEW

**Review by owencjones — COMMENTED** [review]
Note that many files are also reformatted by my formatter, this happens automatically on save, so changed to lines that contain no json commands are incidental

File: darwin/dataset/download_manager.py [hunk 1]
```diff
@@ -3,14 +3,14 @@
 """
```
Comment (owencjones) [inline darwin/dataset/download_manager.py:3]: Many changes that appear on this diff are style changes made by black, the code formatter, not by me.  They are of note, but not relevant to the ticket in play.

File: darwin/dataset/download_manager.py [hunk 2]
```diff
@@ -316,15 +366,16 @@ def _update_local_path(annotation: AnnotationFile, url, local_path):
 
     # we modify raw json, as internal representation does't store all the data
     with annotation.path.open() as file:
-        raw_annotation = json.load(file)
+        raw_annotation = json.loads(file.read())
```
Comment (owencjones) [inline darwin/dataset/download_manager.py:369]: `orjson` is significantly faster, but doesn't expose a `load` method to operate on `TextWrapperIO` functions.

I did plan to write a wrapper for this to expose the same interface, but decided against it for this quick-win, low hanging fruit ticket.

File: darwin/dataset/download_manager.py [hunk 3]
```diff
@@ -316,15 +366,16 @@ def _update_local_path(annotation: AnnotationFile, url, local_path):
...
 
         for slot in raw_annotation["item"]["slots"]:
             for source_file in slot["source_files"]:
                 if source_file["url"] == url:
                     source_file["local_path"] = str(local_path)
 
     with annotation.path.open(mode="w") as file:
-        json.dump(raw_annotation, file, indent=4)
+        op = json.dumps(raw_annotation, json.OPT_INDENT_2)
```
Comment (owencjones) [inline darwin/dataset/download_manager.py]: We can't recreate 4-space indented JSON with `orjson` - it offers 2 spaces, or no spaces.  For the performance gain, this seems fair.

File: darwin/dataset/download_manager.py (hunk 1 above)
Comment (brain-geek) [reply darwin/dataset/download_manager.py:3]: Wouldn't it be better to have separate PR with formatting, to have a clearer PR?

File: .gitignore [hunk 4]
```diff
@@ -161,4 +161,8 @@ cython_debug/
 # conda environment
 environment.yml
 
-.DS_Store
\ No newline at end of file
+.DS_Store
+!.vscode/settings.json
```
Comment (owencjones) [inline .gitignore:165]: Allows commiting vscode files needed without git complaint

File: darwin/dataset/download_manager.py (hunk 1 above)
Comment (owencjones) [reply darwin/dataset/download_manager.py:3]: Addressed now, the settings of the formatter have been changed at @tomvars suggestion, as most were just changing line length anyway.

Might still be one or two, but the main, they're just directly relevent to the code change.

**Review by brain-geek — APPROVED** [review]

File: .gitignore (hunk 4 above)
Comment (owencjones) [reply .gitignore:166]: In reaction to a request from Tom.  The `settings.json` file has been committed for ages, and contains a standard for formatting the files on save etc.  It's already there on `master` - this just stops git from complaining that it's in a commit.

Currently it's a tracked file that is _also_ in the `.gitignore`

Comment (owencjones) [reply .gitignore:166]: RE: the `recommendations.json` file allows us to make suggestions for useful plugins for working with the codebase, but honestly, I don't feel strongly about whether we have it or not.

**Review by rslota — APPROVED** [review]

**Pull request merged**